src/shared/canvas/DataExplorerGraphFlow/DateExplorerGraphFlow.tsx

```diff
@@ -75,25 +83,34 @@ const DataExplorerGraphFlow = () => {
     <div
       className={clsx(
         'data-explorer-resolver',
-        shrinked && 'shrinked',
-        !links.length ? 'no-links' : 'with-links'
+        leftLinks.length && 'left-existed',
+        rightLinks.length && 'right-existed',
+        leftShrinked && 'left-shrinked',
+        rightShrinked && 'right-shrinked',
+        !leftNodes.links.length && !rightNodes.links.length
+          ? 'no-links'
+          : 'with-links'
       )}
       style={
         {
-          '--links-count': shrinked ? 3 : links.length,
+          '--left--links-count': leftShrinked ? 3 : leftNodes.links.length,
+          '--right--links-count': rightShrinked ? 3 : rightNodes.links.length,
         } as CSSProperties
       }
     >
-      <div className="degf__navigation-stack">
-        <NavigationStack />
-      </div>
-      <div className="degf__navigation-back">
-        <NavigationCollapseButton />
-        <NavigationBackButton />
-      </div>
+      {!!leftLinks.length && (
+        <div className="degf__navigation-stack">
+          <NavigationStack key="navigation-stack-left" side="left" />
+        </div>
+      )}
       <div className="degf__content">
         <DataExplorerContentPage />
       </div>
+      {!!rightLinks.length && (
```

Comment:
personal opinion, feel free to ignore. I think the `!!` operator is a bit confusing, especially for someone who don't program often in JS. I find the version `rightLinks.length !== 0` way more explicit and clearer.

Reply:
I think it's a good way to learn the lang, casting to boolean using double NOT operator or Boolean method are clean options